src/ex_run.rs

```diff
@@ -67,7 +67,7 @@ pub fn run_ex_all_tcs(ex_name: &str) -> Result<()> {
 }
 
 pub fn run_ex(ex_name: &str, tc: &str) -> Result<()> {
-    let tc = toolchain::parse_toolchain(tc)?;
+    let tc = tc.parse::<toolchain::Toolchain>()?;
```

Comment:
Glancing through the rest of the code, it looks like all of the occurrences can be replaced with annotating the type of the `let` binding, or just dropped.